### PR TITLE
TUVI-timeout

### PR DESCRIPTION
**Pull request by mihkelkruuse**

### CODE REVIEW

File: dist/client-integration.js [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+        },
+        useSecureTransport: false
+      });
+      return imap.connect().then(() => {
+        expect(imap.client.secureMode).to.be.true;
+      }).then(() => {
+        return imap.close();
+      });
+    });
+    it('should ignore STARTTLS', () => {
```
Comment (littlemaneuver) [inline dist/client-integration.js:127]: this doesn't look right.

1. I would make the lib export dist only to npm, by default folder should be ignored by git
2. this build includes tests? definitely fishy

File: src/client.js [hunk 2]
```diff
@@ -107,7 +107,11 @@ export default class Client {
     clearTimeout(this._idleTimeout)
 
     // propagate the error upwards
-    this.onerror && this.onerror(err)
+    if (!this.onerror) {
```
Comment (soukand) [inline src/client.js:110]: seems really dangerous change

File: dist/client-integration.js (hunk 1 above)
Comment (littlemaneuver) [reply dist/client-integration.js:127]: ok, it seems its a little too late to ignore dist folder
but for build you should run build script https://github.com/emailjs/emailjs-imap-client/blob/master/scripts/build.sh which ingores `-unit` and `-ingegration` files

File: src/client.js (hunk 2 above)
Comment (mihkelkruuse) [reply src/client.js:110]: i agree

**Pull request closed without merging**